### ISSUE TITLE
Backup: an error has occurred

### ISSUE THREAD

**Issue by rik-shaw** [AndBible/and-bible#126]
I am using LineageOS (Android 7.1.2).  When choosing "Administration > Backup > Backup" I am immediately greeted with the message "An error has occurred".  Is it possible that it is a permissions issue or something of that sort?

**Comment by ildar** [comment]
This is hardly a permission problem IMHO.
Could you please upload the logcat? You can do it with plenty of apps and
adb
Please contact me in case of difficulties.

**Comment by rik-shaw** [comment]
@ildar no problem using adb but I need help knowing what logfile you want?  Is this an and-bible specific log or more of a system-level log?

Also note that I have confirmed that and-bible has storage write permissions.  My sdcard is formatted as "internal storage" which by default forces it to be encrypted.  But I don't seem to have trouble manually creating a folder at the root of the sdcard when mounted to my computer (I am using Ubuntu 16.04).

**Comment by ildar** [comment]
No, on Android 7 it's not the case

**Comment by rik-shaw** [comment]
Hmm just seems like a generic error saying "error copying to SD card."  Should I possibly create an empty file at the correct location first or anything else?  Seems strange that I don't see others with this issue.  Thanks for the help on logcat: not too experienced on Android here :-)

```
04-25 15:22:21.315 10043 10077 W System  : ClassLoader referenced unknown path: /system/framework/tcmclient.jar
04-25 15:22:21.317   820   884 D LuxLevels: bright hysteresis constant= 0.1, threshold=29.66884, lux=26.971672
04-25 15:22:21.317   820   884 D LuxLevels: dark hysteresis constant= 0.2, threshold=21.577337, lux=26.971672
04-25 15:22:21.958  9959 10016 D BibleJavascriptIntrfc: Enabling verse long touch selection mode
04-25 15:22:22.008  9959 10016 D BibleJavascriptIntrfc: JS onload
04-25 15:22:22.011  9959 10016 D BibleJavascriptIntrfc: onLoad from js
04-25 15:22:22.014  9959 10016 D BibleJavascriptIntrfc: clear verse positions
04-25 15:22:22.022  9959 10016 D BibleJavascriptIntrfc: Num spans found:135
04-25 15:22:22.028  9959 10016 D BibleJavascriptIntrfc: Num verses found:32
04-25 15:22:22.072  9992  9992 W RenderThread: type=1400 audit(0.0:118): avc: denied { read } for uid=10134 name="reset_count" dev="sysfs" ino=16816 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:sysfs:s0 tclass=file permissive=0
04-25 15:22:22.076  9959  9992 W Adreno-ES20: <rb_perfcounter_gpu_disjoint_status:487>: open failed: errno 13
04-25 15:22:22.084  9959  9992 W Adreno-ES20: <rb_perfcounter_gpu_disjoint_status:487>: open failed: errno 13
04-25 15:22:22.079  9992  9992 W RenderThread: type=1400 audit(0.0:119): avc: denied { read } for uid=10134 name="reset_count" dev="sysfs" ino=16816 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:sysfs:s0 tclass=file permissive=0
04-25 15:22:23.910  9959  9959 D BibleView: Focus changed so start/stop scroll
04-25 15:22:23.910  9959  9959 D BibleView: Pausing tilt to scroll Window [screenNo=1]
04-25 15:22:24.762  1882 10107 I Authzen : [DeviceStateSyncManager] Skipping sync since last one was 20025ms ago.
04-25 15:22:26.289  9959  9992 D OpenGLRenderer: endAllActiveAnimators on 0x97b8fb00 (MenuPopupWindow$MenuDropDownListView) with handle 0x97a8ae40
04-25 15:22:27.050   257   416 I ThermalEngine: Sensor:msm_therm:25000 mC
04-25 15:22:27.624  9959  9992 D OpenGLRenderer: endAllActiveAnimators on 0x97ab4d80 (MenuPopupWindow$MenuDropDownListView) with handle 0x97aeb330
04-25 15:22:28.407  9959  9959 D net.bible.service.common.FileManager: Copying:andBibleDatabase.db
04-25 15:22:28.412  9959  9959 E BackupControl: Error copying database to SD card
04-25 15:22:28.413  9959  9959 D Dialogs : showErrorMesage message:An error has occurred
04-25 15:22:28.875  9959  9992 D OpenGLRenderer: endAllActiveAnimators on 0x9689b500 (MenuPopupWindow$MenuDropDownListView) with handle 0x97aeb860
04-25 15:22:37.055   257   416 I ThermalEngine: Sensor:msm_therm:25000 mC
04-25 15:22:41.801   820   884 D LuxLevels: bright hysteresis constant= 0.1, threshold=33.069256, lux=30.062958
04-25 15:22:41.801   820   884 D LuxLevels: dark hysteresis constant= 0.2, threshold=24.050367, lux=30.062958
```

**Comment by mjdenham** [comment]
The 'Send Report' button should be on the Error dialog.  What version of the app are you using?

**Comment by rik-shaw** [comment]
I have found that the problem was my sd card was used as encrypted "internal storage".  After reformatting it and using it as "portable storage" then the backup will proceed without error.  So I think the problem is that and-bible can't handle a non-portable SD card.  Is there an ability to manually define the backup folder location?  I think this would solve the issue.

**Comment by tuomas2** [comment]
I think I would like this to be implemented the way described in #148. Closing this issue.
